webpack/scenes/Subscriptions/components/SubscriptionsTable/SubscriptionsTableHelpers.js

```diff
@@ -1,22 +1,36 @@
+const getMaxQuantity = (subscription, upstreamAvailable) => {
+  if (upstreamAvailable === -1) {
+    return upstreamAvailable;
+  }
+  return upstreamAvailable + subscription.quantity;
+};
+
 const buildTableRow = (subscription, availableQuantities, updatedQuantity) => {
-  const availableQuantityLoaded = !!availableQuantities;
-  const availableQuantity = availableQuantityLoaded
+  const upstreamAvailableLoaded = !!availableQuantities;
+  const upstreamAvailable = upstreamAvailableLoaded
     ? availableQuantities[subscription.id]
     : null;
 
+  let maxQuantity;
+  if (upstreamAvailableLoaded) {
+    maxQuantity = getMaxQuantity(subscription, upstreamAvailable);
+  }
+
   if (updatedQuantity[subscription.id]) {
     return {
       ...subscription,
       entitlementsChanged: true,
       quantity: updatedQuantity[subscription.id],
-      availableQuantity,
-      availableQuantityLoaded,
+      upstreamAvailable,
+      upstreamAvailableLoaded,
+      maxQuantity,
     };
   }
   return {
     ...subscription,
-    availableQuantity,
-    availableQuantityLoaded,
+    upstreamAvailable,
+    upstreamAvailableLoaded,
+    maxQuantity,
```

Comment:
extra credit since they were like this originally:
Make the shared attributes into another object and destruct that object to DRY it up. i.e.
```javascript
const baseSubscription = {
  ...subscription,
  upstreamAvailable,
  upstreamAvailableLoaded,
  maxQuantity
}
if (updatedQuantity[subscription.id]) {
  return {
    ...baseSubscription,
    entitlementsChanged: true,
    quantity: updatedQuantity[subscription.id],
  }
}
return baseSubscription;
```

You can use the test you wrote to tell if it alters any actual functionality (the test should pass after refactoring)